welkin/models/calendar.py

```diff
@@ -117,3 +127,67 @@ def get(
         return super().get(
             f"{self._client.instance}/calendar/{route}", params=params, *args, **kwargs
         )
+
+
+class WorkHours(Resource):
+    @property
+    def id(self, value=None):
+        if not hasattr(self, "details"):
+            self.details = [{}]
+
+        self.details[0]["workHoursId"] = value
+
+    def create(self, repeating=True):
+        self.startDateTime = datetime.now(tz=timezone.utc)
+        if repeating:
+            self.endDateTime = self.startDateTime.replace(
+                year=self.startDateTime.year + 50
+            )
+
+        return super().post(f"{self._client.instance}/calendar/work-hours")
+
+    def get_all(
+        self,
+        from_date: datetime,
+        to_date: datetime,
+    ):
+
+        params = {
+            "from": from_date,
+            "to": to_date,
+        }
+
+        response = self._client.get(
+            f"{self._client.instance}/calendar/work-hours/", params=params)
+        
+        dict_version = {index: wh for index, wh in enumerate(response[0:-1])}
+        super().update(dict_version)
+        return self
```

Comment:
I think you can delete this, we like to keep the wrapper to just what's implemented by the API so for this case that's the GET on WorkerHours, POST on WorkHours and PUT on Workhours

Reply:
Other pieces of welkin-health have logic for setting up the payload as this one... and many of those are highly used by Zest's project... For example see:

https://github.com/Lightmatter/welkin-health/blob/main/welkin/models/chat.py#L40-L59

Reply:
Right, that's what the WorkerHours Collection is for though. It handles this case of get_all and returns a list of dictionaries of WorkHours. The Collections are for those endpoints that return lists and Resources are for endpoints that return a dictionary. So you should keep that Collection get with the payload setup in favor of this and delete this.

Reply:
Just for further clarity a GET on a Resource is for an ID based get, which the WorkHours lacks, there's only the Collection GET

Reply:
Ed, just added the requested fixes for the get method of this collection. Added three tests for testing  this out, but the put or patch endpoints will be left to do... we really only need to pull this from Welkin. Hope this is good enough.